### PR TITLE
[ADD] sale_margin_mrp: New module

### PR DESCRIPTION
**Pull request by chienandalu**
Recompute kits and manufactured products standard price when the margin is
computed.

cc @Tecnativa TT32040

please review @sergio-teruel @carlosdauden

### CODE REVIEW

**Comment by sergio-teruel** [comment]
@chienandalu fix travis... 
maximum recursion depth exceeded while calling a Python object :smile:

**Comment by chienandalu** [comment]
It should be working now.

**Comment by sergio-teruel** [comment]
:smile_cat: still fails

**Comment by chienandalu** [comment]
Ouch. Silly lint error. Fixed

**Review by sergio-teruel — APPROVED** [review]

**Review by CarlosRoca13 — APPROVED** [review]

**Review by CarlosRoca13 — COMMENTED** [review]
Ups, travis is red

File: sale_margin_mrp/models/sale_order.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+
+class SaleOrderLine(models.Model):
+    _inherit = "sale.order.line"
+
+    @api.onchange("product_id", "product_uom")
+    def product_id_change_margin(self):
+        """Action bom cost computes the standard price from the BoM components.
+        We reproduce here the condition that makes the manual action visible
+        or not in the product form to trigger it automaticaly."""
```
Comment (ivantodorovich) [inline sale_margin_mrp/models/sale_order.py:13]:
```suggestion
        Override. Action bom cost computes the standard price from the BoM components.
        We reproduce here the condition that makes the manual action visible
        or not in the product form to trigger it automaticaly.
```

Nitpicking here: When overwritting, it's better not to overwrite the original method's docstring

File: sale_margin_mrp/models/sale_order.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+        if (
+            self.product_id and self.product_id.bom_count
+            and self.product_id.valuation != "real_time"
+        ):
+            product = self.product_id.with_context(
+                force_company=self.order_id.company_id.id
+            ).sudo()
+            product.write({"standard_price": product._get_price_from_bom()})
```
Comment (ivantodorovich) [inline sale_margin_mrp/models/sale_order.py:21]: Is this really what you want?

It's a bit invasive to write on the product everytime a product is added to the SO I believe.


Shouldn't this be writing on the sale.order.line's `purchase_price` instead? If so you could probably override `_compute_margin` instead, and achieve the same in a cleaner way

**Comment by pedrobaeza** [comment]
Closing as not finally desired.

**Pull request closed without merging**